src/main/java/book/store/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package book.store.onlinebookstore.service.impl;
+
+import book.store.onlinebookstore.dto.userdto.UserRegistrationRequestDto;
+import book.store.onlinebookstore.dto.userdto.UserResponseDto;
+import book.store.onlinebookstore.exceptions.RegistrationException;
+import book.store.onlinebookstore.mapper.UserMapper;
+import book.store.onlinebookstore.model.User;
+import book.store.onlinebookstore.repository.UserRepository;
+import book.store.onlinebookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto authenticate(UserRegistrationRequestDto userRegistrationRequestDto)
+            throws RegistrationException {
+        String userEmail = userRegistrationRequestDto.getEmail();
+        User user = userRepository.findByEmail(userEmail).orElseThrow(() ->
+                new RegistrationException("User by email " + userEmail + " is not already exists"));
+        return userMapper.toDto(user);
+    }
```

Comment:
1. Naming wasn't fixed
2. Logic is not correct. We should throw exception in case user with such email exists or create new one in case email is free (this method is create operation for user by fact but with some additional logic)
```suggestion
    @Override
    public UserResponseDto register(UserRegistrationRequestDto userRegistrationRequestDto)
            throws RegistrationException {
        String userEmail = userRegistrationRequestDto.getEmail();
        Optional<User> existingUser = userRepository.findByEmail(userEmail);
        if (existingUser.isPresent()) {
                throw new RegistrationException("User by email " + userEmail + " already exists");
        }
        //map userRegistrationRequestDto to user
        //save user to db
        return userMapper.toDto(savedUser);
    }
```

Reply:
ehh ok, txh, in first version i have logic something like a that

Reply:
and i thought that i have to change name only in controller layer, sorry